be/src/storage/lake/gc.cpp

```diff
@@ -2,131 +2,189 @@
 
 #include "storage/lake/gc.h"
 
+#include <rapidjson/error/en.h>
+#include <rapidjson/reader.h>
+#include <rapidjson/writer.h>
+
 #include <algorithm>
 #include <ctime>
 #include <unordered_map>
 
 #include "common/config.h"
 #include "fs/fs.h"
+#include "fs/rapidjson_stream_adapter.h"
 #include "storage/lake/filenames.h"
 #include "storage/lake/join_path.h"
 #include "storage/lake/location_provider.h"
 #include "storage/lake/tablet_manager.h"
 #include "storage/lake/tablet_metadata.h"
+#include "util/raw_container.h"
 
 namespace starrocks::lake {
 
-static std::string format_time(time_t ts) {
-    struct tm tm {};
-    char buffer[32];
-    // Format: 2009-06-06 20:20:00 UTC
-    auto len = std::strftime(buffer, sizeof(buffer), "%Y.%m.%d %H:%M:%S UTC", gmtime_r(&ts, &tm));
-    return {buffer, len};
+static const char* const kOrphanSegmentKey = "orphan_segments";
+
+struct OrphanSegmentHandler {
```

Comment:
suggest wrap this with an anonymous namespace so it will never pollute external namespaces accidentally.

Reply:
Updated, PTAL